common/display/src/main/scala/uk/ac/wellcome/display/serialize/AggregationRequestSerializer.scala

```diff
@@ -0,0 +1,59 @@
+package uk.ac.wellcome.display.serialize
+
+import com.fasterxml.jackson.core.{JsonParser, JsonProcessingException}
+import com.fasterxml.jackson.databind.module.SimpleModule
+import com.fasterxml.jackson.databind.{DeserializationContext, JsonDeserializer}
+import uk.ac.wellcome.display.models._
+
+class AggregationRequestParsingException(msg: String)
+    extends JsonProcessingException(msg: String)
+
+class AggregationRequestDeserializer
+    extends JsonDeserializer[List[AggregationRequest]] {
+
+  override def deserialize(
+    p: JsonParser,
+    ctx: DeserializationContext): List[AggregationRequest] = {
+
+    // The value that comes through here is similar to `["foo,bar"]`
+    // So we jump into the array and then grab that text
+    p.nextValue()
+    val commaSeparatedString = p.getText()
+
+    // We create a (key: String, agg: Option[Aggregation]) tuple so we can error nicely telling people which values are invalid
+    val validations: Map[Boolean, List[(String, Option[AggregationRequest])]] =
+      commaSeparatedString
+        .split(",")
+        .toList
+        .map(_.trim)
+        .map(key => (key, AggregationRequest(key))) groupBy {
+        case (_, maybeAggregationRequest) => maybeAggregationRequest.isEmpty
+      }
```

Comment:
Not a biggy, but potentially things might be a bit simpler here if we have a `Either[AggregationRequest, String]` to represent valid / invalid values: when it is valid we are not interested in the `key`, whereas when it is invalid the second value is always `None`, so modelling as a tuple does not make as much sense to me.

Reply:
Yea - `Either`s are definitely what we should use for validation here.
Changed the `AggregationRequest.apply` to  return `Either[InvalidAggregationRequest, AggregationRequest]` which tidies things up quite nicely - thanks!